code/modules/wiremod/shell/module.dm

```diff
@@ -0,0 +1,246 @@
+		RegisterSignal(mod, COMSIG_ITEM_EQUIPPED, PROC_REF(equip_check))
+		wearer.set_output(mod.wearer)
+		var/modules_list = list()
+		for(var/obj/item/mod/module/module in mod.modules)
+			if(module.module_type != MODULE_PASSIVE)
+				modules_list += module.name
+		module_to_select.possible_options = modules_list
+		if (module_to_select.possible_options.len)
+			module_to_select.set_value(module_to_select.possible_options[1])
+	else if(istype(old_loc, /obj/item/mod/control))
+		UnregisterSignal(old_loc, list(COMSIG_MOD_MODULE_SELECTED, COMSIG_ITEM_EQUIPPED))
+		UnregisterSignal(old_loc, COMSIG_MOD_DEPLOYED)
+		UnregisterSignal(old_loc, COMSIG_MOD_RETRACTED)
+		UnregisterSignal(old_loc, COMSIG_MOD_TOGGLED)
+		UnregisterSignal(old_loc, COMSIG_MOD_MODULE_ADDED)
+		UnregisterSignal(old_loc, COMSIG_MOD_MODULE_REMOVED)
+		selected_module.set_output(null)
+		wearer.set_output(null)
+		deployed.set_output(FALSE)
+		activated.set_output(FALSE)
+
+/obj/item/circuit_component/mod_adapter_core/proc/on_module_select(datum/source, obj/item/mod/module/module)
+	SIGNAL_HANDLER
+	selected_module.set_output(module.name)
+	on_module_selected.set_output(COMPONENT_SIGNAL)
+
+/obj/item/circuit_component/mod_adapter_core/proc/on_module_changed()
+	SIGNAL_HANDLER
+	var/modules_list = list()
+	for(var/obj/item/mod/module/module in attached_module.mod.modules)
+		if(module.module_type != MODULE_PASSIVE)
+			modules_list += module.name
+	module_to_select.possible_options = modules_list
+	if (module_to_select.possible_options.len)
+		module_to_select.set_value(module_to_select.possible_options[1])
+
+/obj/item/circuit_component/mod_adapter_core/proc/on_mod_part_toggled()
+	SIGNAL_HANDLER
+	var/string_list = list()
+	var/is_deployed = TRUE
+	for(var/obj/item/part as anything in attached_module.mod.get_parts())
+		if(part.loc == attached_module.mod)
+			is_deployed = FALSE
+		else
+			var/part_name = "Undefined"
+			if(istype(part, /obj/item/clothing/head/mod))
+				part_name = "Helmet"
+			if(istype(part, /obj/item/clothing/suit/mod))
+				part_name = "Chestplate"
+			if(istype(part, /obj/item/clothing/gloves/mod))
+				part_name = "Gloves"
+			if(istype(part, /obj/item/clothing/shoes/mod))
+				part_name = "Boots"
+			string_list += part_name
+	deployed_parts.set_output(string_list)
+	deployed.set_output(is_deployed)
+	on_deploy.set_output(COMPONENT_SIGNAL)
+
+/obj/item/circuit_component/mod_adapter_core/proc/on_mod_toggled()
+	SIGNAL_HANDLER
+	activated.set_output(attached_module.mod.active)
+	on_toggle_finish.set_output(COMPONENT_SIGNAL)
+
+/obj/item/circuit_component/mod_adapter_core/proc/equip_check()
+	SIGNAL_HANDLER
+	if(!attached_module.mod?.wearer)
+		return
+	wearer.set_output(attached_module.mod.wearer)
+
+/obj/item/circuit_component/equipment_action/mod
+	display_name = "MOD action"
+	desc = "Represents an action the user can take when wearing the MODsuit."
+	required_shells = list(/obj/item/mod/module/circuit)
+*/
```

Comment:
why is all of this commentetd out?

Reply:
This is desirable that I want to include it, but our circuit code is incompatible with it.

The incompatibilities are minor, but I do not know enough about circuits to make it workable. Ive spent atleast 3 hours on a separate branch with no progress.